xds/src/main/java/io/grpc/xds/BootstrapperImpl.java

```diff
@@ -326,14 +323,10 @@ private static ChannelCredentials parseChannelCredentials(List<Map<String, ?>> j
         throw new XdsInitializationException(
             "Invalid bootstrap: server " + serverUri + " with 'channel_creds' type unspecified");
       }
-      switch (type) {
-        case "google_default":
-          return GoogleDefaultChannelCredentials.create();
-        case "insecure":
-          return InsecureChannelCredentials.create();
-        case "tls":
-          return TlsChannelCredentials.create();
-        default:
+      XdsCredentialsProvider provider =  XdsCredentialsRegistry.getDefaultRegistry()
+          .getProvider(type);
+      if (provider != null) {
+        return provider.getChannelCredentials(channelCreds);
```

Comment:
The "config" key of `channelCreds` is what should be passed (with an empty map if the field wasn't present).

Reply:
Done